shell/platform/linux/public/flutter_linux/fl_value.h

```diff
@@ -363,31 +363,35 @@ double fl_value_get_float(FlValue* value);
 
 /**
  * fl_value_get_string:
- * @value: an #FlValue of type #FL_VALUE_TYPE_STRING
+ * @value: an #FlValue of type #FL_VALUE_TYPE_STRING or #FL_VALUE_TYPE_NULL.
  *
  * Gets the UTF-8 text contained in @value. Calling this with an #FlValue
- * that is not of type #FL_VALUE_TYPE_STRING is a programming error.
+ * that is not of type #FL_VALUE_TYPE_STRING or #FL_VALUE_TYPE_NULL is a
+ * programming error.
  *
- * Returns: a UTF-8 encoded string.
+ * Returns: a UTF-8 encoded string or %NULL if the type is #FL_VALUE_TYPE_NULL.
```

Comment:
I'm torn on whether this one should exist. I understand the convenience, but from an API perspective it feels more consistent to me to say "it's okay to call collection getters on null" than "it's okay to call things that return pointers on null, which happens to be all the collections and then exactly one single-value type".

Having an inconsistency at all isn't ideal, but having dealt with the pain of this with rapidjson I do think it's worth paying that price, but I feel like leaving strings out is probably less confusing, even if also a little less convenient. I think it'll be easier to remember that any time you get down to a leaf type you need to check for null, rather than thinking of strings and numbers/bools as behaving fundamentally different.

Reply:
I haven't landed this because I expected more cases where channels had null set for values, but I haven't encountered them. I think it's more reasonable to expect the API user to explicitly check for null in all cases if they care about it rather than guessing which types support it. I'm finding the most reasonable pattern for parsing messages is to basically assert the things that should be exactly what you expect::

```c++
if (fl_value_get_type(value) != FL_VALUE_TYPE_LIST) {
  g_warning("Everything is terrible");
  return;
}
do_the_thing(fl_value_get_list(value));
```

and treat anything that might be optional/wrong as `nullptr` if they're not what you expect:
```c++
const char *text = nullptr;
if (fl_value_get_type(value) == FL_VALUE_TYPE_STRING)
  text = fl_value_get_string(value);
```